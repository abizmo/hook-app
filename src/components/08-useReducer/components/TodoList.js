import React from 'react'
import PropTypes from 'prop-types'

import TodoListItem from './TodoListItem'

const TodoList = ({ onDeleteTodo, onToggleTodo, todos }) => (
  <ul className="list-group">
    {
      todos.map(({id, todo, done}) => (
        <TodoListItem
          key={id}
          done={done}
          id={id}
          onDeleteTodo={onDeleteTodo}
          onToggleTodo={onToggleTodo}
          todo={todo}
        />
      ))
    }
  </ul>
)

TodoList.propTypes = {
  onDeleteTodo: PropTypes.func.isRequired,
  onToggleTodo: PropTypes.func.isRequired,
  todos: PropTypes.arrayOf(PropTypes.shape({
    done: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    todo: PropTypes.string.isRequired,
  }))
}

export default TodoList
