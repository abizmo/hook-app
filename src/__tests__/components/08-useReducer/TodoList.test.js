import { shallow } from 'enzyme'

import TodoList from "../../../components/08-useReducer/components/TodoList"
import todos from '../../../fixtures/demoTodos'


describe('TodoList tests', () => {
  const handleDeleteTodo = jest.fn()
  const handleToggleTodo = jest.fn()
  const wrapper = shallow(
    <TodoList
      onDeleteTodo={handleDeleteTodo}
      onToggleTodo={handleToggleTodo}
      todos={todos}
    />
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('TodoListItem')).toHaveLength(todos.length)
    expect(typeof wrapper.find('TodoListItem').at(0).prop('onDeleteTodo')).toBe('function')
    expect(typeof wrapper.find('TodoListItem').at(0).prop('onToggleTodo')).toBe('function')
  })
})
