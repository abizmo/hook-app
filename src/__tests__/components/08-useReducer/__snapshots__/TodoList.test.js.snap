// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList tests should render and match snapshot 1`] = `
<ul
  className="list-group"
>
  <TodoListItem
    done={true}
    id={1}
    key="1"
    onDeleteTodo={[MockFunction]}
    onToggleTodo={[MockFunction]}
    todo="Comprar pan"
  />
  <TodoListItem
    done={false}
    id={2}
    key="2"
    onDeleteTodo={[MockFunction]}
    onToggleTodo={[MockFunction]}
    todo="Ir al colegio"
  />
</ul>
`;
